### ISSUE TITLE
[v2] Cannot convert value to AST: IntValue{value=1}

### ISSUE THREAD

**Issue by blacksmoke26** [jimkyndemeyer/js-graphql-intellij-plugin#217]
When I click:
![image](https://user-images.githubusercontent.com/425744/54041310-32fae680-41e9-11e9-9e02-9e34c0e32090.png)

Throws error:
```
9:24 PM	GraphQL Introspection Error
'Cannot convert value to AST: IntValue{value=1}
Retry
```

and Retry does the same, v1 works great but this breaks everything.

**Comment by jimkyndemeyer** [comment]
Hi Junaid.

Thanks for testing v2.

I believe that this is similar to #216

I've seen this error being thrown by `graphql-java` which v2 uses to build up the schema. It occurs when the default value of a field argument doesn't follow the GraphQL specification which states that it should be encoded using GraphQL language and not JSON which I've seen GraphQL frameworks do, including GitHub's API.

Once the introspection runs, the "Query result" tab in the GraphQL tool window will contain the result of the introspection query, including the `defaultValue` fields. Please review these fields. The important thing is that it's the GraphQL parser that needs to understand the default values. From the error message you're seeing my guess is that your endpoint introspection returned something similar to:

```
{
  "defaultValue": "{intFieldExample:\"1\"}"
}
```

In this case the number should not have been quoted.

I hope that helps. Let me know whether changing your default values to align with the GraphQL spec fixes the issue. If you are already on-spec, it's likely that a fix is needed for `graphql-java`.

**Comment by blacksmoke26** [comment]
Is there any solution to deal with it? Maybe I should wait for the next version?

**Comment by blacksmoke26** [comment]
Sure, Will try!

**Comment by jimkyndemeyer** [comment]
Hi Junaid.

The developer experience here could definitely be improved as it's not clear from the error notification what's going on and how to fix it.

To clarify, I think the one of the following is happening:

- The GraphQL endpoint serializes defaults in a non-spec way when returned in the `defaultValue` introspection field. In this case you should raise a ticket with framework that the GraphQL endpoint is based on. Or, if you're developing the endpoint, you may have provided a wrong default value using the GraphQL framework API, e.g. a string where an int is expected. In that case you can fix it.
- There could also be an error in this plugin or with graphql-java. In that case, please attach the introspection result to this issue, and I'll attempt to fix it.

A workaround for now, to let you work with v2, is to manually copy the introspection query result into a new `schema.json` file and edit the invalid `defaultValue` to be valid GraphQL input. You can use the JSON file as your schema definition by specifying `"schemaPath": "schema.json"` in your `.graphqlconfig`

**Comment by jimkyndemeyer** [comment]
The error thrown is in https://github.com/graphql-java/graphql-java/blob/516a0c0d8121e3a79b1b83b0069b51526793c5f5/src/main/java/graphql/language/AstValueHelper.java#L118

@blacksmoke26 It would help if you could attach your introspection query result to this issue. I need to reproduce in order to improve the developer experience with better error handling.

**Comment by blacksmoke26** [comment]
Thanks, So far I went through my schema, it seems that args default values for Array / Object / Custom Types (e.g., JSON) causing the error:

Please skip no existings types (e.g., `PageInfo`, `PositiveInt`, `JSON`, ...) Those are already in schema.
```graphql
type CountriesConnection {
  nodes: [Country!]
  pageInfo: PageInfo!
  totalCount: Int
}

enum CountriesOrderBy {
  NATURAL
  ID_ASC
  ID_DESC
  NAME_ASC
  NAME_DESC
  SLUG_ASC
  SLUG_DESC
  PRIMARY_KEY_ASC
  PRIMARY_KEY_DESC
}

type Country {
  id: PositiveInt!
  name: String!
  code: String!
  locale: String!
  currency: String!
  language: String!
  slug: String!
  coordinates: [Float!]
  provinces(pager: PagerOptions = {page: false, limit: false, type: CURSOR},
    sortOrder: [ProvincesOrderBy!] = [NAME_ASC], findOptions: JSON): ProvincesConnection
}

type Query {
  countries(
      pager: PagerOptions = {page: false, limit: false, type: CURSOR},
      sortOrder: [CountriesOrderBy!] = [NAME_ASC],
      findOptions: JSON
    ) : CountriesConnection
}
```

```js
{
  "data": {
    "__schema": {
      "queryType": {
        "name": "Query"
      },
      "mutationType": {
        "name": "Mutation"
      },
      "subscriptionType": null,
      "types": [{
        "kind": "OBJECT",
        "name": "Query",
        "description": "",
        "fields": [{
          "name": "countries",
          "description": "",
          "args": [{
            "name": "pager",
            "description": "",
            "type": {
              "kind": "INPUT_OBJECT",
              "name": "PagerOptions",
              "ofType": null
            },
            "defaultValue": "{page: 1, limit: 10, type: CURSOR}" // <--- Error (1) => JSON default value
          }, {
            "name": "sortOrder",
            "description": "",
            "type": {
              "kind": "LIST",
              "name": null,
              "ofType": {
                "kind": "NON_NULL",
                "name": null,
                "ofType": {
                  "kind": "ENUM",
                  "name": "CountriesOrderBy",
                  "ofType": null
                }
              }
            },
            "defaultValue": "[NAME_ASC]"


          }, {
            "name": "findOptions",
            "description": "",
            "type": {
              "kind": "SCALAR",
              "name": "JSON",
              "ofType": null
            },
            "defaultValue": null
          }],
          "type": {
            "kind": "OBJECT",
            "name": "CountriesConnection",
            "ofType": null
          },
          "isDeprecated": false,
          "deprecationReason": null
        }],
        "inputFields": null,
        "interfaces": [],
        "enumValues": null,
        "possibleTypes": null
      }]
    }
  }
}
```

Schema introspection query result:
[schema.zip](https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/files/2948775/schema.zip)

My schema has no issues, Validated already.

**Comment by jimkyndemeyer** [comment]
Thanks for the schema. I was able to reproduce and make custom scalars work with introspection.

The fix is in the 2.0.0-alpha-8 which I'll release later today.

**Comment by blacksmoke26** [comment]
Issue solved. Thank You!
